Engine/Component/ComponentScript.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ComponentScript.h"
+#include "Main/Application.h"
+#include "Module/ModuleRender.h"
+#include "Main/GameObject.h"
+#include "Module/ModuleScriptManager.h"
+#include "Script/Script.h"
+#include "imgui.h"
+
+ComponentScript::ComponentScript() : Component(nullptr, ComponentType::SCRIPT)
+{
+
+}
+ComponentScript::ComponentScript(GameObject * owner, std::string script_name) : Component(owner, ComponentType::SCRIPT)
+{
+	script = App->scripts->CreateResourceScript(script_name, owner);
+	App->scripts->scripts.push_back(this);
+}
+
+ComponentScript::~ComponentScript()
+{
+}
```

Comment:
You could put the destructor as default and erase these lines of code.

Reply:
Done